src/components/ui4/PromoCardsUi4.tsx

```diff
@@ -0,0 +1,109 @@
+import { asArray } from 'cleaners'
+import { asPromoCard2, PromoCard2 } from 'edge-info-server/types'
+import * as React from 'react'
+import { Platform } from 'react-native'
+import { getBuildNumber, getVersion } from 'react-native-device-info'
+
+import { getCountryCodeByIp } from '../../actions/AccountReferralActions'
+import { config } from '../../theme/appConfig'
+import { NavigationBase } from '../../types/routerTypes'
+import { fetchInfo } from '../../util/network'
+import { EdgeAnim } from '../common/EdgeAnim'
+import { useTheme } from '../services/ThemeContext'
+import { CarouselUi4 } from './CarouselUi4'
+import { PromoCardUi4 } from './PromoCardUi4'
+
+interface Props {
+  navigation: NavigationBase
+  screenWidth: number
+}
+
+export const PromoCardsUi4 = (props: Props) => {
+  const { navigation, screenWidth } = props
+  const theme = useTheme()
+
+  const [promos, setPromos] = React.useState<PromoCard2[]>([])
+
+  // Check for PromoCard2 from info server:
+  React.useEffect(() => {
+    fetchInfo(`v1/promoCards2/${config.appId ?? 'edge'}`)
+      .then(async res => {
+        if (!res.ok) {
+          console.error(await res.text())
+          return
+        }
+        const infoData = await res.json()
+        const infoPromoCards = asArray(asPromoCard2)(infoData)
+
+        const buildNumber = getBuildNumber()
+        const countryCode = await getCountryCodeByIp()
+        const osType = Platform.OS.toLowerCase()
+        const osVersion = Platform.OS === 'ios' ? Platform.Version : `api${Platform.Version}`
+        const version = getVersion()
+        const currentDate = new Date()
+
+        setPromos(
+          infoPromoCards.filter(infoPromoCard => {
+            const { appVersion, minBuildNum, maxBuildNum, exactBuildNum, countryCodes, excludeCountryCodes, osTypes, osVersions, startIsoDate, endIsoDate } =
+              infoPromoCard
+
+            // Validate app version
+            // Ignore everything else if build version is specified and mismatched.
+            if (exactBuildNum != null && exactBuildNum !== buildNumber) return false
+
+            // Ignore min/max if app version is specified and mismatched.
+            if (appVersion != null && appVersion !== version) return false
+
+            // Look at min/max only if exact build or app version is not specified.
+            const isMinVersion = minBuildNum == null || minBuildNum <= buildNumber
+            const isMaxVersion = maxBuildNum == null || maxBuildNum >= buildNumber
+            if (!isMinVersion || !isMaxVersion) return false
```

Comment:
I won't require a fixup but this should have just been
```
if (minBuildNum == null || minBuildNum > buildNumber) return false
if (maxBuildNum == null || maxBuildNum < buildNumber) return false
```

Get rid of those darn `isMinVersion` type variables. They're unneeded.